source/extensions/filters/udp/dns_filter/dns_parser.cc

```diff
@@ -196,13 +196,17 @@ bool DnsMessageParser::parseDnsObject(DnsQueryContextPtr& context,
   // Only QR == 0 and questions without any answer RRs are expected
   if (!(context->header_.flags.qr == 0 && context->header_.answers == 0 &&
         context->header_.authority_rrs == 0 && context->header_.additional_rrs == 0)) {
-    ENVOY_LOG(debug, "One or more of Answer, Authority, and Additional RRs present in the query",
-              "Inverse query is not supported");
+    ENVOY_LOG(
+        debug,
+        "One or more of Answers [{}], Authority [{}], and Additional [{}] RRs present in the query."
+        "Inverse query is not supported",
+        static_cast<int>(context->header_.answers),
+        static_cast<int>(context->header_.authority_rrs),
+        static_cast<int>(context->header_.additional_rrs));
```

Comment:
This is not what I meant (I meant adding a filter stat/metric for this failure case that can be monitored), but if you want to keep that that is fine.